pkg/reconciler/managed/reconciler.go

```diff
@@ -681,6 +721,11 @@ func NewReconciler(m manager.Manager, of resource.ManagedKind, o ...ReconcilerOp
 		ro(r)
 	}
 
+	if err := m.Add(&r.driftRecorder); err != nil {
+		r.log.Info("unable to register drift recorder with controller manager", "error", err)
```

Comment:
Should this be a debug log? What would a typical user do with/about this log when they saw it?

Reply:
Would even promote it to an error if I could without logging interface. This signals that something is broken. It's not debug output.

Reply:
Alternative: we add an error as return value as a breaking change.

Reply:
Some rationale as to why we don't have error log level: https://dave.cheney.net/2015/11/05/lets-talk-about-logging

Reply:
I think my original comment still stands though - how would a user fix this? What would cause this? In the spirit of the better errors guide - what does "unable to register drift recorder" mean?

Keeping at info level is fine with me if it's meaningful and actionable to the user. Channeling the good errors guide, I don't personally think I'd know what "unable to register drift recorder" meant if I hadn't read this PR, or what I should do about it.